lib/charliecloud.py

```diff
@@ -894,6 +898,11 @@ def parse(class_, s):
                                    parser="earley", propagate_positions=True)
       s = s.replace("%", "/").replace("+", ":")
       hint="https://hpc.github.io/charliecloud/faq.html#how-do-i-specify-an-image-reference"
+      while (s.find("$") > -1): #variable substitution
+         start = s.find("$")
+         end = s.find(":", start)
+         str = s[start, end - 1]
+         s.replace(str, self.dict.get(str[1: -1]))
```

Comment:
I think you can use the existing `variables_sub()` here? You might need to move it to `charliecloud.py` and update existing callers.

Reply:
`variables_sub()` isn't able to find the values. Should I try to tweak `variables_sub()` and make it work for what I want it to do or make something new?

Reply:
That's surprising to me. Yes, I think changing `variables_sub()` is the way to go.